### PR TITLE
Add devtools to view menu

### PR DESCRIPTION
**Pull request by mtsgrd**

### CODE REVIEW

**Comment by Qix-** [comment]
@mtsgrd yes though let's check with @PavelLaptev since he's doing something similar right now 😅

**Comment by PavelLaptev** [comment]
Ignore me, I was just playing around with the 'view' menu. Let's do @mtsgrd stuff first 🙂

File: crates/gitbutler-tauri/src/menu.rs [hunk 1]
```diff
@@ -31,10 +31,30 @@ pub async fn menu_item_set_enabled(
...
+                        CustomMenuItem::new("view/devtools", "Developer Tools").into(),
+                    ];
+                    custom_items.extend(sm.inner.items);
+                    extended_menu = extended_menu
+                        .add_submenu(Submenu::new("View", Menu::with_items(custom_items)));
+                } else {
+                    extended_menu = extended_menu.add_submenu(sm.clone());
+                }
+            }
+            _ => continue,
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs]:
```suggestion
        if let MenuEntry::Submentu(sm) = item {
                if sm.title == "View" {
                    let mut custom_items = vec![
                        disabled_menu_item("view/settings", "Project Settings"),
                        CustomMenuItem::new("view/devtools", "Developer Tools").into(),
                    ];
                    custom_items.extend(sm.inner.items);
                    extended_menu = extended_menu
                        .add_submenu(Submenu::new("View", Menu::with_items(custom_items)));
                } else {
                    extended_menu = extended_menu.add_submenu(sm.clone());
                }
            }
```

Probably not a 'valid' diff but you can match a single variant like this.

File: crates/gitbutler-tauri/src/menu.rs [hunk 2]
```diff
@@ -31,10 +31,30 @@ pub async fn menu_item_set_enabled(
...
+    let mut extended_menu = Menu::new();
+    for item in menu.items {
+        match item {
+            MenuEntry::Submenu(sm) => {
+                if sm.title == "View" {
+                    let mut custom_items = vec![
+                        disabled_menu_item("view/settings", "Project Settings"),
+                        CustomMenuItem::new("view/devtools", "Developer Tools").into(),
+                    ];
+                    custom_items.extend(sm.inner.items);
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs]: Instead of creating a vec here, just `push` them directly.

File: crates/gitbutler-tauri/src/menu.rs [hunk 3]
```diff
@@ -31,10 +31,30 @@ pub async fn menu_item_set_enabled(
...
+                    extended_menu = extended_menu
+                        .add_submenu(Submenu::new("View", Menu::with_items(custom_items)));
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs]: If you're already pushing the new items to the existing menu then why add a new submenu?

**Comment by Qix-** [comment]
FWIW here's the code I sent to @PavelLaptev; you can of course just do a loop + `if` instead to clean things up a bit.

```rust
pub fn build(package_info: &PackageInfo) -> Menu {
    let mut menu = Menu::os_default(&package_info.name).add_submenu(Submenu::new(
        "Project",
        Menu::with_items([disabled_menu_item("project/settings", "Project Settings")]),
    ));

    // Try to find the View menu and attach the dev tools item
    let view_menu = menu
        .items
        .iter_mut()
        .find(|item| match item {
            MenuEntry::CustomItem(_) => false,
            MenuEntry::Submenu(submenu) => submenu.title == "View",
            MenuEntry::NativeItem(_) => false,
        });

    if let Some(MenuEntry::Submenu(view_menu)) = view_menu {
        view_menu
            .inner
            .items
            .push(CustomMenuItem::new("app/dev-tools", "Toggle Developer Tools").into());
    }

    menu
}
```

File: crates/gitbutler-tauri/src/menu.rs [hunk 4]
```diff
@@ -44,6 +60,9 @@ fn disabled_menu_item(id: &str, title: &str) -> MenuEntry {
 }
 
 pub fn handle_event<R: Runtime>(event: &WindowMenuEvent<R>) {
+    if event.menu_item_id() == "view/devtools" {
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs:66]: This doesn't match the string above. Think it needs to be `app/dev-tools` here, or alternatively change the above to match this one.

**Pull request merged**